File: src/Entity/Commande.php
<?php

namespace App\Entity;

use App\Repository\CommandeRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use DateTimeImmutable;

#[ORM\Table(name: "commande", indexes: [
    new ORM\Index(name: "idPanier", columns: ["idPanier"])
])]
#[ORM\Entity(repositoryClass: CommandeRepository::class)]
class Commande
{
    #[ORM\Column(name: "idCommande", type: "integer", nullable: false)]
    #[ORM\Id]
    #[ORM\GeneratedValue(strategy: "IDENTITY")]
    private $idcommande;

    #[ORM\Column(name: "dateCommande", type: Types::DATETIME_IMMUTABLE, nullable: false)]
    private \DateTimeInterface $datecommande;

    #[ORM\ManyToOne(targetEntity: Panier::class)]
    #[ORM\JoinColumn(name: "idPanier", referencedColumnName: "idPanier")]
    private $idpanier;

    public function getIdcommande(): ?int
    {
        return $this->idcommande;
    }

    public function getDatecommande(): \DateTimeInterface
    {
        return $this->datecommande;
    }

    public function setDatecommande(\DateTimeInterface $datecommande): static
    {
        $this->datecommande = $datecommande;

        return $this;
    }

    public function getIdpanier(): ?Panier
    {
        return $this->idpanier;
    }

    public function setIdpanier(?Panier $idpanier): static
    {
        $this->idpanier = $idpanier;

        return $this;
    }
}

File: src/main/java/entities/Commande.java
package entities;

import java.util.Date;

public class Commande {
    int idCommande;
    Panier panier;
    Date dateCommande;

    public Commande()
    {

    }
    public Commande(int idCommande,Panier panier,Date dateCommande)
    {
        this.idCommande=idCommande;
        this.panier=panier;
        this.dateCommande=dateCommande;
    }

    public int getIdCommande() {
        return idCommande;
    }

    public void setIdCommande(int idCommande) {
        this.idCommande = idCommande;
    }

    public Panier getPanier() {
        return panier;
    }

    public void setPanier(Panier panier) {
        this.panier = panier;
    }

    public Date getDateCommande() {
        return dateCommande;
    }

    public void setDateCommande(Date dateCommande) {
        this.dateCommande = dateCommande;
    }
}
